File: Problema1/factura.py
class Factura:

    def __init__(self, numero_pieza, descripcion_pieza, cantidad, precio_articulo):
        self.numero_pieza = numero_pieza
        self.descripcion_pieza = descripcion_pieza

        if cantidad > 0:
            self.cantidad = cantidad
        else:
            self.cantidad = 0

        if precio_articulo > 0:
            self.precio_articulo = precio_articulo
        else:
            self.precio_articulo = 0.0

    def obtener_monto_factura(self):
        return self.cantidad * self.precio_articulo


factura1 = Factura("A001", "Martillo", 3, 150.0)

print("Monto de la factura: $", factura1.obtener_monto_factura())

File: Problema1/Factura.cs
using System;

class Factura
{
    private string numeroPieza;
    private string descripcionPieza;
    private int cantidad;
    private double precioArticulo;

    public Factura(string numeroPieza, string descripcionPieza, int cantidad, double precioArticulo)
    {
        this.numeroPieza = numeroPieza;
        this.descripcionPieza = descripcionPieza;

        if (cantidad > 0)
            this.cantidad = cantidad;
        else
            this.cantidad = 0;

        if (precioArticulo > 0)
            this.precioArticulo = precioArticulo;
        else
            this.precioArticulo = 0.0;
    }

    public double ObtenerMontoFactura()
    {
        return cantidad * precioArticulo;
    }
}

class Program
{
    static void Main()
    {
        Factura factura1 = new Factura("A001", "Martillo", 3, 150.0);

        Console.WriteLine("Monto de la factura: $" + factura1.ObtenerMontoFactura());
    }
}
